pkg/counter/counter.go

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package counter
+
+// Counter tracks references for comparable .
+//
+// No thread safety is provided within this structure, the user is expected to
+// handle concurrent access to this structure if it is used from multiple
+// threads.
+type Counter[T comparable] map[T]int
+
+// Add increments the reference count for the specified key.
+func (c Counter[T]) Add(key T) (changed bool) {
```

Comment:
Won't it be more readable like this?
```
func (c Counter[T]) Add(key T) bool {
        value, exists := c[key]
        c[key] = value + 1
        return !exists
}
```
(for me `changed` sounds like the value was changed)